### PR TITLE
SITES-22965: Add function to set category filter type

### PR DESCRIPTION
**Pull request by alwinjoseph02**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Review by GiovanniReza — APPROVED** [review]

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/categorylist/FeaturedCategoryListImpl.java [hunk 1]
```diff
@@ -106,6 +108,8 @@ private void initModel() {
         // After the identifier type has been determined, the specific list will be used further
         List<String> categoryIdentifiers = new ArrayList<>();
         assetOverride = new HashMap<>();
+        String categoryIdType = null;
+        String categoryFilterType;
```
Comment (LSantha) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/categorylist/FeaturedCategoryListImpl.java:112]: Could you check if you need this categoryFilterType variable? It looks like it's serving basically the same purpose as categoryIdType. If so, please keep categoryIdType only.

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/categorylist/FeaturedCategoryListImpl.java [hunk 2]
```diff
@@ -76,6 +77,7 @@ public class FeaturedCategoryListImpl extends DataLayerComponent implements Feat
     private static final String CATEGORY_IDENTIFIER = "categoryId";
     private static final String ASSET_PROP = "asset";
     private static final String ITEMS_PROP = "items";
+    private static final String CATEGORY_IDENTIFIER_TYPE = "categoryIdType";
```
Comment (LSantha) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/categorylist/FeaturedCategoryListImpl.java:80]: You may want to put this on the public interface like here: https://github.com/adobe/aem-core-cif-components/pull/1011/files#diff-5e1c4c18538024ef6a86682d30b765717aeea3d4af9372f76de7b34190c67afdR44

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java [hunk 3]
```diff
@@ -66,6 +66,12 @@ public abstract class AbstractCategoriesRetriever extends AbstractRetriever {
      */
     protected List<String> identifiers;
 
+    /**
+     * Filter type that should be used when fetched. usually uid but we can define it explicitly in the implementation
+     * specific and should be checked in subclass implementations.
+     */
+    protected String filterType;
```
Comment (LSantha) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java:73]: Here also categoryIdType is a better name and is consistent with the setter method bellow.

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java [hunk 4]
```diff
@@ -93,6 +99,16 @@ public void setIdentifiers(List<String> identifiers) {
         this.identifiers = identifiers;
     }
 
+    /**
+     * Set the Category filter type which will using during fetch. Categories are retrieved using
+     * this filter type if not set then it will use UID.
+     *
+     * @param filterType Filter Type
```
Comment (LSantha) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java:106]: Parameter name: categoryIdType

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java [hunk 5]
```diff
@@ -93,6 +99,16 @@ public void setIdentifiers(List<String> identifiers) {
...
+     */
+    public void setCategoryIdType(String filterType) {
+        this.filterType = filterType;
```
Comment (LSantha) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java:109]: this.categoryIdType = categoryIdType

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoryRetriever.java [hunk 6]
```diff
@@ -118,6 +129,16 @@ public void setIdentifier(String identifier) {
         this.identifier = identifier;
     }
 
+    /**
+     * Set the Category filter type which will using during fetch. Categories are retrieved using
+     * this filter type if not set then it will use UID.
+     *
+     * @param filterType Filter Type
```
Comment (LSantha) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoryRetriever.java:136]: Here too: categoryIdType

**Comment by LSantha** [comment]
@alwinjoseph02 , LGTM!

**Review by LSantha — APPROVED** [review]

**Pull request merged**